src/gogolica/gen/model/method.clj

```diff
@@ -0,0 +1,84 @@
+(ns gogolica.gen.model.method
+  "Functions to access the data about API model methods."
+  (:gen-class)
+  (:require [clojure.string :as str]
+            [cheshire.core :as json]
+            [me.raynes.fs :as fs]
+            [camel-snake-kebab.core :refer :all]
+            [gogolica.gen.uri-template :as uri-template]))
+
+(defn media-upload?
+  "Checks whether the given method supports uploading media."
+  [method]
+  (and (-> method :supportsMediaUpload)
+       (-> method :mediaUpload)))
+
+(defn media-upload
+  "Get the details on what protocols are supported and what kind of media
+   can be uploaded for given method."
+  [method]
+  (-> method :mediaUpload))
+
+(defn media-download?
+  "Checks whether a given method provides downloadable media."
+  [method]
+  (-> method :supportsMediaDownload boolean))
+
+(defn media-download-service?
+  [method]
+  (-> method :useMediaDownloadService boolean))
+
+(defn parameters
+  [method]
+  (merge
+   (-> method :parameters)
+   (when (media-download? method)
+     {:alt {:position "query"}})))
+
+(defn path-parameters
+  [method]
+  (->> method parameters
+       (filter #(-> % val :location (= "path")))))
+
+(defn query-parameters
+  [method]
+  (->> method parameters
+       (filter #(-> % val :location (= "query")))))
+
+(defn path
+  "The path template of a given method.
+   Represented as a vector of strings and symbols."
+  [method]
+  (-> method :path
+      (uri-template/parse
+       (->> method
+            path-parameters
+            keys
+            (mapv name)))))
```

Comment:
This thing with double arrows is a bit confusing. Let's refactor it with like a let